anitya/lib/backends/maven.py

```diff
@@ -62,11 +62,17 @@ def get_versions(cls, project):
             url = project.homepage
         else:
             coordinates = project.version_url or project.name
-            if ':' not in coordinates:
+            match = re.match(r'([^:]+):([^:]+)', coordinates)
```

Comment:
Since this regular expression is not dynamically generated and will probably used fairly frequently, it's probably worth putting it at the module level and compiling it:
```
# At the module level
#: A regular expression used to validate that Maven ``version_url`` fields are actually coordinates,
#: which are just two strings, a group ID and an artifact ID separated by a colon. 
coordinates_regex = re.compile(r'([^:]+):([^:]+)')

...

# right here
match = coordinates_regex.match(coordinates)
```